packages/core/test/lib/integrations/inboundfilters.test.ts

```diff
@@ -1,522 +1,314 @@
-import { InboundFilters } from '../../../src/integrations/inboundfilters';
-
-let inboundFilters: any;
-
-describe('InboundFilters', () => {
-  beforeEach(() => {
-    inboundFilters = new InboundFilters();
-  });
-
-  describe('shouldDropEvent', () => {
-    it('should drop when error is internal one', () => {
-      inboundFilters._isSentryError = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should drop when error is ignored', () => {
-      inboundFilters._isIgnoredError = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should drop when url is denied', () => {
-      inboundFilters._isDeniedUrl = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should drop when url is not allowed', () => {
-      inboundFilters._isAllowedUrl = () => false;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should drop when url is not denied, but also not allowed', () => {
-      inboundFilters._isDeniedUrl = () => false;
-      inboundFilters._isAllowedUrl = () => false;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should drop when url is denied and allowed at the same time', () => {
-      inboundFilters._isDeniedUrl = () => true;
-      inboundFilters._isAllowedUrl = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(true);
-    });
-
-    it('should not drop when url is not denied, but allowed', () => {
-      inboundFilters._isDeniedUrl = () => false;
-      inboundFilters._isAllowedUrl = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(false);
-    });
-
-    it('should not drop when any of checks dont match', () => {
-      inboundFilters._isIgnoredError = () => false;
-      inboundFilters._isDeniedUrl = () => false;
-      inboundFilters._isAllowedUrl = () => true;
-      expect(inboundFilters._shouldDropEvent({}, inboundFilters._mergeOptions())).toBe(false);
-    });
-  });
-
-  describe('isSentryError', () => {
-    const messageEvent = {
-      message: 'captureMessage',
-    };
-    const exceptionEvent = {
-      exception: {
-        values: [
-          {
-            type: 'SyntaxError',
-            value: 'unidentified ? at line 1337',
-          },
-        ],
+import { EventProcessor } from '@sentry/types';
+
+import { InboundFilters, InboundFiltersOptions } from '../../../src/integrations/inboundfilters';
+
+/** JSDoc */
+function createInboundFiltersEventProcessor(
```

Comment:
> It's also not totally clear to me as a reader (yet) why what we want is the event processor rather than a dummy instance of the integration.

This is also a function of the unit testing of integrations being kind of inconsistent.

Will add a doc string